internal/domain/review_suggestion.go

```diff
@@ -0,0 +1,27 @@
+package domain
+
+import (
+	"github.com/invopop/jsonschema"
+)
+
+type ReviewSuggestion struct {
+	Comment  string `json:"comment"`
+	Position int    `json:"position"`
+}
+
+type ReviewSuggestionsResponse struct {
+	ReviewSuggestions []ReviewSuggestion `json:"review_suggestions"`
+}
+
+func GenerateSchema[T any]() interface{} {
+	reflector := jsonschema.Reflector{
+		AllowAdditionalProperties: false,
+		DoNotReference:            true,
+	}
```

Comment:
Consider using `sync.Once` or similar patterns to ensure thread-safe initialization and avoid the performance hit from initialization every time the variable is accessed in a concurrent scenario.